src/workflow/ArcanistDiffWorkflow.php

```diff
@@ -1187,6 +1189,61 @@ private function shouldAmend() {
     return true;
   }
 
+/* -(  Custom Check-Prompt-Resolve scripts  ) ------------------------------- */
+
+  /**
+   * @task checkPromptResolveScripts
+   */
+  private function runCheckPromptResolveScripts() {
+    $cprScriptDefs = $this->getConfigurationManager()->getConfigFromAnySource(
+          'uber.differential.check_prompt_resolve'
+        );
+    if (is_null($cprScriptDefs)) {
+      return;
+    }
+    foreach ($cprScriptDefs as $cprDef) {
+      $fNames = ['check-script', 'prompt', 'default-to', 'resolve-script'];
+      foreach ($fNames as $fieldName) {
+        if (is_null($cprDef[$fieldName])) {
+          throw new ArcanistUsageException(
+                      pht("Malformed uber.differential.check_prompt_resolve ".
+                          "record: %s (must have \"%s\" field)",
+                          $cprDef, $fieldName));
+        }
+      }
+      $checkScript = $cprDef['check-script'];
+      $prompt = $cprDef['prompt'];
+      $defaultsToYes = $cprDef['default-to'];
+      $resolveScript = $cprDef['resolve-script'];
+      list ($err, $stdout, $stderr) = exec_manual('%s', $checkScript);
+      if ($err == 0) {
+        continue; // Nothing to fix
+      }
+      $doResolve = phutil_console_confirm(
+                            $prompt, $default_no = !$defaultsToYes);
+      list ($err, $stdout, $stderr) = exec_manual('%s', $resolveScript);
```

Comment:
probably pipe `$stdout` and `$stderr` to console since these would could include full build logs, etc.

Reply:
👍